src/test/java/com/cdancy/bitbucket/rest/features/TasksApiLiveTest.java

```diff
@@ -121,15 +120,31 @@ public void testGetTaskOnError() {
         assertThat(instance).isNotNull();
         assertThat(instance.errors().isEmpty()).isFalse();
     }
-    
+
+    @Test
+    public void testTaskStatusUpdate() {
+        final CreateTask createTask = CreateTask.create(commentId, taskComment);
+        final Task instanceAlreadyOpen = api().create(createTask);
+
+        final Task instanceNowResolved = api().update(instanceAlreadyOpen.id(),"RESOLVED");
+        assertThat(instanceNowResolved).isNotNull();
+        assertThat(instanceNowResolved.errors().isEmpty()).isTrue();
+        assertThat(instanceNowResolved.state()).isEqualTo("RESOLVED");
+
+        final Task instanceReopened = api().update(instanceNowResolved.id(),"OPEN");
+        assertThat(instanceReopened).isNotNull();
+        assertThat(instanceReopened.errors().isEmpty()).isTrue();
+        assertThat(instanceReopened.state()).isEqualTo("OPEN");
+    }
+
```

Comment:
Lets also add another test to check when things fail. Maybe pass in a bogus taskId and ensure that the `errors()` object gets properly populated.

Reply:
I have added an on failure test in 4ba5d22